### PR TITLE
DOC: Replace Anaconda with Miniconda + newer install process in dev docs

### LINKED ISSUE #13724: DOC: Replace Anaconda with Miniconda in dev doc
https://docs.astropy.org/en/latest/development/workflow/virtual_pythons.html

We should encourage the use of Miniconda over Anaconda. We do not want user to get bloated base env before they even start doing dev work on astropy.

* Replace Anaconda with Miniconda.
* Maybe even add a note to discourage the use of full Anaconda install.

### PR DESCRIPTION
**Pull request by BSGalvan**
### Description

PR to nudge new developers to use `miniconda` instead of Anaconda, in order to reap the benefits of a more minimal `base`.

Also added were updates to dev docs for the newer install process (see discussion at #13724).

Several minor typos are also corrected.

Should close #13724 once merged.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
Hello and thank you!

Please do not use `git pull`, as that inserts merge commits within an unmerged pull request. Can you please do the following?

1. `git fetch upstream main` + `git rebase upstream/main`
2. Squash all the remaining commits into 1 commit using interactive git rebase.

Please let me know if you need assistance with this. Thanks!

**Comment by BSGalvan** [comment]
Hope that is looks cleaner now.
Thank you for the instructions!

File: docs/development/workflow/development_workflow.rst [hunk 1]
```diff
@@ -168,6 +168,13 @@ our coding style standards. This can be done by installing ``pre-commit`` in the
 of your astropy repository by running::
 
     pip install pre-commit
+
+Or if you prefer `conda`_::
+
+    conda install pre-commit
```
Comment (pllim) [inline docs/development/workflow/development_workflow.rst:174]: @WilliamJamieson or @nstarman , would you recommend pre-commit from conda or should it always be pip?

File: docs/development/workflow/virtual_pythons.rst [hunk 2]
```diff
@@ -28,15 +28,25 @@ Astropy and other Python packages.
...
-* If you use the Anaconda/Conda Python distribution you must use the
+* If you wish to use the `miniconda`_ Python distribution, you must use the
   `conda`_ command to make and manage your virtual environments.
 
-* If you do not use Anaconda you can use `virtualenv`_ and the conda-like
+.. note::
+
+    `miniconda`_ is a minimal flavor of the popular Anaconda Python
+    distribution, containing only ``conda``, Python, other useful packages
+    (such as pip, requests etc.) and their dependencies in the
```
Comment (pllim) [inline docs/development/workflow/virtual_pythons.rst:43]:
```suggestion
    (such as ``pip``, ``requests``, etc.) and their dependencies in the
```

File: docs/development/workflow/virtual_pythons.rst [hunk 3]
```diff
@@ -28,15 +28,25 @@ Astropy and other Python packages.
...
+    ``base``/``root`` environment.
```
Comment (pllim) [inline docs/development/workflow/virtual_pythons.rst:44]: `root` has been deprecated a long time, so don't think we have to mention it.
```suggestion
    ``base`` environment.
```

File: docs/development/workflow/development_workflow.rst (hunk 1 above)
Comment (WilliamJamieson) [reply docs/development/workflow/development_workflow.rst:174]: I don't use conda so I do not know how well this works.

File: docs/development/workflow/virtual_pythons.rst [hunk 4]
```diff
@@ -136,31 +147,24 @@ environment.
...
-      `virtualenvwrapper command documentation`_.
-
 * `conda`_:
-    + Make an environment called ``ENV`` with all of the packages in your main
-      Anaconda environment::
+    + Make an environment called ``ENV`` with all of the packages in your ``base``
+      Miniconda environment::
 
-        conda create -n ENV anaconda
+        conda create -n ENV --clone base
```
Comment (pllim) [inline docs/development/workflow/virtual_pythons.rst:154]: No need to clone base.

```suggestion
        conda create -n ENV
```

File: docs/development/workflow/virtual_pythons.rst [hunk 5]
```diff
@@ -173,6 +177,28 @@ environment.
 
         conda uninstall astropy
 
+    + Depending on your development use case, you may want to install
+      additional packages into this environment in order to carry out tests,
+      build documentation, extend specific additional features etc. See
+      :ref:`testing-dependencies`, :ref:`builddocs` and :ref:`Requirements for
```
Comment (pllim) [inline docs/development/workflow/virtual_pythons.rst:183]:
```suggestion
      :ref:`testing-dependencies`, :ref:`builddocs`, and :ref:`Requirements for
```

File: docs/development/workflow/virtual_pythons.rst [hunk 6]
```diff
@@ -173,6 +177,28 @@ environment.
...
+         mkvirtualenv --system-site-packages ENV
+
+    + Omit the option ``--system-site-packages`` to create an environment
+      without the Python packages installed in your default Python environment.
+    + Environments created with `virtualenvwrapper`_ always include `pip`_
+      and `setuptools <https://setuptools.readthedocs.io>`_ so that you
+      can install packages within the virtual environment.
+    + More details and examples are in the
+      `virtualenvwrapper command documentation`_.
+
```
Comment (pllim) [inline docs/development/workflow/virtual_pythons.rst:200]:
```suggestion
```

File: docs/nitpick-exceptions [hunk 7]
```diff
@@ -106,6 +106,8 @@ py:obj Artist
 py:obj BboxBase
 py:obj Text
 py:obj text
+## required for describing conda base env
+py:obj base
```
[the remark this answers is not in the document]
Comment (BSGalvan) [reply docs/nitpick-exceptions:110]: Yup, just rechecked and leaving this out doesn't throw the error sphinx-build was throwing earlier. Must have been something unrelated 🤷🏽

**Comment by pllim** [comment]
@parejkoj , do you want to have a look?

@BSGalvan , when review is done, the commits will have to be squashed (again) into one commit. Thanks for your patience!

**Comment by BSGalvan** [comment]
> @BSGalvan , when review is done, the commits will have to be squashed (again) into one commit. Thanks for your patience!

Sure thing!

File: docs/development/workflow/development_workflow.rst (hunk 1 above)
Comment (nstarman) [reply docs/development/workflow/development_workflow.rst:174]: pre-commit from conda works great (provided of course that one is working in a conda environment).

Comment (nstarman) [reply docs/development/workflow/development_workflow.rst:174]: If one is using conda, all things should be installed via conda if they can. pip + conda doesn't always play nice.

**Comment by pllim** [comment]
@BSGalvan , since there are no other comments, would you mind squash all commits into one and rebase? Thanks!

**Comment by BSGalvan** [comment]
@pllim I have squashed and rebased, but the rewritten history isn't showing up here. Any cause for concern?

**Comment by pllim** [comment]
@BSGalvan , yes... Did you do a force push?

**Comment by BSGalvan** [comment]
> @BSGalvan , yes... Did you do a force push?

Yeah, I did a force push. The branch on my fork also shows that it is only one commit ahead of `astropy/main`, with all the changes suggested during review.

What I did earlier today was this:
- Synchronized my fork with the current state of `astropy/main`, since it had fallen behind (caused a `merge` commit, but I thought I could fix that during the `rebase`)
- Started fresh by cloning once again.
- Did a `rebase` and squashed the commits into one.

**Comment by pllim** [comment]
> Started fresh by cloning once again

What do you mean by this? Cloning how?

**Comment by pllim** [comment]
Also check your remote URLs. Your fork didn't register what you just did today.

Screenshot 2022-10-26 082221

**Comment by pllim** [comment]
If you have multiple local checkouts, make sure you are in the correct one. Just trying to narrow down the problem here...

**Comment by BSGalvan** [comment]
Okay, so I initially suspected that the `rebase` didn't update the commit date, which is why the branch activity wasn't updated (and neither was this PR). I went and amended the latest commit, which [now shows up in the branch activity](https://github.com/BSGalvan/astropy/branches) but the PR still hasn't updated.
I'm making sure to use the correct local checkout so I'm at my wits' end as to what is going wrong.

**Comment by pllim** [comment]
Something is very wrong now. It pulled unrelated commits. Let me try for you. Hope you don't mind.

**Comment by pllim** [comment]
Yeah, my push doesn't seem to do anything. Could be a lag or maybe github is acting weird because you set your PR branch as the default branch of your fork.

**Comment by BSGalvan** [comment]
Done.

**Comment by pllim** [comment]
Alas, that didn't help. Not sure what happened to your fork but in the interest of getting your changes in in time for feature freeze, I went ahead and opened https://github.com/astropy/astropy/pull/13908 from your clean-up branch. Please have a look over there. Thanks!

**Pull request closed without merging**